indra/llcommon/llsd.cpp

```diff
@@ -840,6 +840,23 @@ void LLSD::clear()						{ Impl::assignUndefined(impl); }
 
 LLSD::Type LLSD::type() const			{ return safe(impl).type(); }
 
+bool LLSD::isEmpty() const
+{
+    Type t = type();
+    if (t == TypeUndefined)
+        return true;
+    if (t == TypeBinary)
+        return !asBoolean();
+    if (t == TypeInteger)
+        return !asInteger();
+    if (t == TypeReal)
+        return !asReal();
+    if (t == TypeString || t == TypeMap || t == TypeArray)
+        return !size();
+    return false;
```

Comment:
I suggest using a `switch` since Type is an enum. Ideally to use `safe(impl).isEmpty()` to make sure no type gets missed